testsuite/tests/multicluster/conftest.py

```diff
@@ -0,0 +1,184 @@
+        request.addfinalizer(route.delete)
+        route.commit()
+        routes[client] = route
+    return routes
+
+
+@pytest.fixture(scope="module")
+def hostname(gateways, hub_openshift, exposer, blame) -> Hostname:
+    """Exposed Hostname object"""
+    hostname = exposer.expose_hostname(blame("hostname"), gateways[hub_openshift])
+    return hostname
+
+
+@pytest.fixture(scope="module")
+def exposer(request, hub_openshift) -> Exposer:
+    """Expose using DNSPolicy"""
+    exposer = DNSPolicyExposer(hub_openshift)
+    request.addfinalizer(exposer.delete)
+    exposer.commit()
+    return exposer
+
+
+@pytest.fixture(scope="module")
+def base_domain(exposer):
+    """Returns preconfigured base domain"""
+    return exposer.base_domain
+
+
+@pytest.fixture(scope="module")
+def wildcard_domain(base_domain):
+    """
+    Wildcard domain of openshift cluster
+    """
+    return f"*.{base_domain}"
+
+
+@pytest.fixture(scope="module")
+def dns_policy(blame, hub_openshift, gateways, module_label):
+    """DNSPolicy fixture"""
+    policy = DNSPolicy.create_instance(
+        hub_openshift, blame("dns"), gateways[hub_openshift], labels={"app": module_label}
+    )
+    return policy
+
+
+@pytest.fixture(scope="module")
+def tls_policy(blame, hub_openshift, gateways, module_label, cluster_issuer):
+    """TLSPolicy fixture"""
+    policy = TLSPolicy.create_instance(
+        hub_openshift,
+        blame("tls"),
+        parent=gateways[hub_openshift],
+        issuer=cluster_issuer,
+        labels={"app": module_label},
+    )
+    return policy
+
+
+@pytest.fixture(scope="module")
+def dns_policies(openshifts, dns_policy) -> dict[OpenShiftClient, DNSPolicy]:
+    """Creates DNSPolicy for each Openshift server"""
+    return generate_policies(openshifts, dns_policy)
+
+
+@pytest.fixture(scope="module")
+def tls_policies(openshifts, tls_policy) -> dict[OpenShiftClient, TLSPolicy]:
+    """Creates TLSPolicy for each Openshift server"""
+    return generate_policies(openshifts, tls_policy)
+
+
+@pytest.fixture(scope="module")
+def client(hostname, gateways):  # pylint: disable=unused-argument
+    """Returns httpx client to be used for requests"""
+    time.sleep(180)  # it takes a bit for the lets encrypt secret to be actually created
+    root_cert = resources.files("testsuite.resources").joinpath("letsencrypt-stg-root-x1.pem").read_text()
+    client = hostname.client(verify=Certificate(certificate=root_cert, chain=root_cert, key=""))
+    yield client
+    client.close()
+
+
+@pytest.fixture(scope="module", autouse=True)
+def commit(request, routes, dns_policies, tls_policies):  # pylint: disable=unused-argument
+    """Commits all policies before tests"""
+    components = [*dns_policies.values(), *tls_policies.values()]
+    for component in components:
+        request.addfinalizer(component.delete)
+        component.commit()
+    for component in components:
+        component.wait_for_ready()
```

Comment:
I tried running using kua-stage-single, kua-multi-1, kua-multi-2 and here I needed to increase the timeout. TLSPolicies did not get ready in time, it was due to Certificate was not issued quick enough. If I used 180s timeout the test passed ok.

Reply:
It seems fixed

Reply:
Yes, it works now, thanks!